zap/src/main/java/org/zaproxy/zap/core/scanner/InputVector.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.core.scanner;
+
+/**
+ * {@code InputVector} class wraps the parameters which are used to modify the {@code HttpMessage}.
+ * It is specifically used for updating multiple parameters of {@code HttpMessage}
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ */
+public class InputVector {
+    private int position;
+    private String name;
+    private String value;
+    private PayloadFormat payloadFormat;
```

Comment:
It needs to have the format for name and value, since they can be set raw/escaped separately.

Reply:
so are you suggesting to have 2 payloadformat parameters?